### PR TITLE
Core/Client - Fix creating alias/NFT addresses from OutputID and use new JSON formatting rules

### LINKED ISSUE #350: Client - JSON response with 0x prefix
## Description

New restful API response JSON formatting rules:
* Number: any field that contains a number and the representation is below 53 bits
* Plain string encoding: uint64 values such as IOTA amount.
* Hex encoded string with 0x prefix:
  *  uint256 values such as amount of native tokens, some foundry fields.
  * ByteArray and ByteArray[N] values such as hashes, addresses, transactionId, data payload, etc.

for example, an output response
```
{
  "messageId": "0x9cd745ef6800c8e8c80b09174ee4b250b3c43dfa62d7c6a4e61f848febf731a0",
  "transactionId": "0x1ee46e19f4219ee65afc10227d0ca22753f76ef32d1e922e5cbe3fbc9b5a5298",
  "outputIndex": 1,
  "isSpent": false,
  "milestoneIndexBooked": 1234567,
  "milestoneTimestampBooked": 1643207146,
  "ledgerIndex": 946704,
  "output": {
    "type": 3,
    "amount": "1000",
    "unlockConditions": [
      {
        "type": 0,
        "address": {
          "type": 0,
          "pubKeyHash": "0x8eaf87ac1f52eb05f2c7c0c15502df990a228838dc37bd18de9503d69afd257d"
        }
      }
    ]
  }
}
```

### LINKED ISSUE #376: Core - invalid Alias/NFT addresses
## Description

Alias/NFT addresses are invalid that created from: 
* `alias_address_from_output`
* `nft_address_from_output`

the input data of `iota_blake2b_sum` should be the binary data of OutputID, not a string.
```
iota_blake2b_sum(output_id, IOTA_OUTPUT_ID_BYTES, addr->address, NFT_ID_BYTES);
```

### PR DESCRIPTION
**Pull request by matjazv**
# Description of change

- Fix #376 
- Fix #350 
- Fix some invalid variable types `uint64_t` -> `uint32_t`
- Added new query parameter (`aliasAddress`) for API calls

## Type of change

- Bugfix (a non-breaking change which fixes an issue)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

All local and tangle unit tests passed.

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests using CTest that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/client/api/json_parser/common.c [hunk 1]
```diff
@@ -62,21 +62,22 @@ int json_parser_common_address_deserialize(cJSON *json_obj, char const *const js
...
 {
   "type": 16,
-  "nftId": "a3308d6262ae9921f46aa3308d6262ae9921f46a"
+  "nftId": "0xa3308d6262ae9921f46aa3308d6262ae9921f46a"
 }
 */
 cJSON *json_parser_common_address_serialize(address_t *address) {
-  char addr_str[BIN_TO_HEX_STR_BYTES(ADDRESS_MAX_BYTES)] = {};
+  char addr_str[BIN_TO_HEX_STR_BYTES(ADDRESS_MAX_BYTES) + JSON_HEX_ENCODED_STRING_PREFIX_LEN] = {};
+  memcpy(addr_str, "0x", JSON_HEX_ENCODED_STRING_PREFIX_LEN);
```
Comment (oopsmonk) [inline src/client/api/json_parser/common.c]: maybe support `prefix` parameter in `hex_2_bin` and `bin_2_hex`? so don't need to copy the prefix everywhere, like
```
int hex_2_bin(char const str[], size_t str_len, char const *prefix, byte_t bin[], size_t bin_len) 
int bin_2_hex(byte_t const bin[], size_t bin_len, char const *prefix, char str_buf[], size_t buf_len)
```
the prefix parameter is optional, set to NULL for regular conversion

```
bin_2_hex(address->address, ED25519_PUBKEY_BYTES, "0x", addr_str, sizeof(addr_str))
```
even define "0x" as `JSON_HEX_ENCODED_STRING_PREFIX`

File: src/client/api/json_parser/inputs/inputs.c [hunk 2]
```diff
@@ -54,7 +54,7 @@ int json_inputs_deserialize(cJSON *inputs_obj, utxo_inputs_list_t **inputs) {
 }
 
 cJSON *json_inputs_serialize(utxo_inputs_list_t *inputs) {
-  char tx_id_str[BIN_TO_HEX_STR_BYTES(IOTA_TRANSACTION_ID_BYTES)] = {};
+  char tx_id_str[BIN_TO_HEX_STR_BYTES(IOTA_TRANSACTION_ID_BYTES) + JSON_HEX_ENCODED_STRING_PREFIX_LEN] = {};
```
Comment (oopsmonk) [inline src/client/api/json_parser/inputs/inputs.c]: it's long declaration, should define another macros for hex with prefix?
```suggestion
  char tx_id_str[BIN_TO_HEX_STR_WITH_PREFIX_BYTES(IOTA_TRANSACTION_ID_BYTES)] = {};
```

**Review by oopsmonk — CHANGES_REQUESTED** [review]

File: src/client/api/json_parser/common.c (hunk 1 above)
Comment (matjazv) [reply src/client/api/json_parser/common.c]: Thank you for this code improvement suggestion. It was done in this way.

File: src/client/api/json_parser/inputs/inputs.c (hunk 2 above)
Comment (matjazv) [reply src/client/api/json_parser/inputs/inputs.c]: Additional macro was introduced `JSON_STR_WITH_PREFIX_BYTES(x)`.

**Review by oopsmonk — APPROVED** [review]

**Review by rebinmannady93 — APPROVED** [review]

**Pull request merged**